pkg/network/common/egressip.go

```diff
@@ -350,6 +350,20 @@ func (eit *EgressIPTracker) DeleteNetNamespaceEgress(vnid uint32) {
 	})
 }
 
+// NodeInOtherEgressUse verifies if the node - which is ID'd by its nodeIP - still
+// has a reference to it from any **other** egress IP on the cluster, excluding
+// the egress IP provided in the argument.
+func (eit *EgressIPTracker) NodeInOtherEgressUse(egressIP, nodeIP string) bool {
+	for eip, egressIPInfo := range eit.egressIPs {
+		if eip != egressIP {
+			if egressIPInfo.nodes[0].nodeIP == nodeIP {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
hm... it feels messy to do this check here... it would be nicer if, say, `vxlanMonitor.AddNode` and `vxlanMonitor.RemoveNode` were changed to be `AddEgressIP`/`RemoveEgressIP`, and took both the egress IP and the node IP as args, and kept track of all of the known egress IPs associated with a node, and only unmonitored the node when the last egress IP was removed.

Reply:
Done, could you please take another look @danwinship ?